### PR TITLE
Added deques module, deprecating queues

### PR DESCRIPTION
**Pull request by endragor**
Deque can work as a Stack or a Queue or both. The module is pretty much a copy of queues module with renamings and two procedures added (addFirst, popLast).

Proc names for main operations are: addFirst/popFirst/peekFirst and addLast/popLast/peekLast.

### CODE REVIEW

File: lib/pure/collections/deques.nim [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+  emptyCheck(deq)
+  dec deq.count
+  deq.tail = (deq.tail - 1) and deq.mask
+  result = deq.data[deq.tail]
+  deq.data[deq.tail] = default(type(result))
+
+proc `$`*[T](deq: Deque[T]): string =
+  ## Turn a deque into its string representation.
+  result = "["
+  for x in items(deq):  # Don't remove the items here for reasons that don't fit in this margin.
```
Comment (Araq) [inline lib/pure/collections/deques.nim]: Funny, but you can use a multi line comment instead.

Comment (endragor) [inline lib/pure/collections/deques.nim]: That came from `queues`. So I'm not sure what the reasons were :)

Comment (endragor) [inline lib/pure/collections/deques.nim]: I guess it only mattered long time ago, so I removed it.

File: lib/pure/collections/deques.nim [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+  ## via the ``in`` operator. It is the equivalent of ``deq.find(item) >= 0``.
+  ##
+  ## .. code-block:: Nim
+  ##   if x in q:
+  ##     assert q.contains x
+  for e in deq:
+    if e == item: return true
+  return false
+
+template expandIfNeeded(deq) =
```
Comment (endragor) [inline lib/pure/collections/deques.nim]: Done.

**Pull request merged**